ctapipe_io_lst/calibration.py

```diff
@@ -211,15 +211,20 @@ def __init__(self, subarray, config=None, parent=None, **kwargs):
         if self.calibration_path is not None:
             self.mon_data = self._read_calibration_file(self.calibration_path)
 
-    def apply_drs4_corrections(self, event: ArrayEventContainer):
+    def apply_drs4_corrections(self, event: LSTArrayEventContainer):
         self.update_first_capacitors(event)
 
-        for tel_id in event.r0.tel:
+        for tel_id, r0 in event.r0.tel.items():
             r1 = event.r1.tel[tel_id]
             # fill r1 waveform with a copy of the r0 converted to float32
             # float32 can represent all values of uint16 exactly, so this
             # does not loose precision.
-            r1.waveform = event.r0.tel[tel_id].waveform.astype(np.float32)
+            if r1.waveform is None:
+                r1.waveform = r0.waveform
+            else:
+                r1.waveform = r1.waveform
```

Comment:
why do you make a=a type asignment?

Reply:
unnecessary, correct.

Reply:
It was `waveform = r1.waveform` before, when refactoring, I didn't realize that the statement was no superflous